### PR TITLE
Add goldenrun only and missing only flags

### PR DESCRIPTION
**Pull request by berkayurun**
This PR adds two flags to ARCHIE. The "goldenrun-only" flag only executes the goldenrun and creates backup information. The "missing-only" flag simulates missing faults that has not been simulated and saved to the given HDF5 file.

### CODE REVIEW

File: controller.py [hunk 1]
```diff
@@ -464,6 +483,60 @@ def read_backup(hdf5_file):
...
+
+def get_not_simulated_faults(faultlist, simulated_faults):
+    missing_faultlist = []
+
+    for faultconfig in faultlist:
+        h = hashlib.sha256()
+        for fault in faultconfig["faultlist"]:
+            h.update(str(fault).encode())
+
+        if not h.digest() in simulated_faults:
```
Comment (lukasauer) [inline controller.py:534]: If `h.digest()` is in `simulated_faults` we should test if the actual configuration matches the simulated one.

File: controller.py [hunk 2]
```diff
@@ -610,9 +630,27 @@ def controller(
...
+        total=len(faultlist), desc="Simulating faults", disable=not len(faultlist)
+    )
     itter = 0
     while 1:
+        if stop_signal_received.value == 1:
+            clogger.info(
+                "Stop signal received, finishing the current write operation..."
+            )
+
+            while p_logger.is_alive():
```
Comment (lukasauer) [inline controller.py:712]: We should not need to run this in a while loop since `join()` is blocking.

File: controller.py [hunk 3]
```diff
@@ -47,6 +48,24 @@
...
 
+stop_signal_received = Value("i", 0)
+
+
+def signal_handler(signum, frame):
+    global stop_signal_received
+    stop_signal_received.value = 1
+
+
+def register_signal_handlers(clogger, queue_output, p_logger):
```
Comment (lukasauer) [inline controller.py:59]: Please remove the unused function arguments

File: controller.py [hunk 4]
```diff
@@ -487,6 +487,60 @@ def read_backup(hdf5_file):
...
+                        fault["fault_wildcard"],
+                    )
+                    for fault in exp.faults.iterrows()
+                ],
+            }
+
+            config_string = ""
+            for fault in simulated_exp["faultlist"]:
+                config_string += str(fault)
+            simulated_faults_hash[config_string] = simulated_exp
```
Comment (lukasauer) [inline controller.py:523]: The value `simulated_exp` is never used. Is there a more efficient data structure that we could use here? Looks good otherwise.

Comment (aewag) [reply controller.py:523]:
```
            simulated_faults_hash[config_string] = simulated_exp.keys()
```

**Review by lukasauer — APPROVED** [review]

**Pull request merged**